dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ConjureBodySerDe.java

```diff
@@ -80,6 +80,13 @@
                 emptyContainerDeserializer,
                 BinaryEncoding.OPTIONAL_MARKER);
         this.emptyBodyDeserializer = new EmptyBodyDeserializer(errorDecoder);
+        // Class unloading: Not supported, Jackson keeps strong references to the types
+        // it sees: https://github.com/FasterXML/jackson-databind/issues/489
+        this.serializers =
+                Caffeine.newBuilder().build(token -> new EncodingSerializerRegistry<>(defaultEncoding, token));
+        this.deserializers = Caffeine.newBuilder()
+                .build(token -> new EncodingDeserializerRegistry<>(
+                        encodingsSortedByWeight, errorDecoder, emptyContainerDeserializer, token));
```

Comment:
What's the rough heap retention footprint per serializer/deserializer? Should we bound these to avoid growing indefinitely? Wiring config through to here seems tricky, so wondering if we just set an arbitrary max size limit to keep top N, or heuristically max weight it by max heap?

```suggestion
        this.serializers = Caffeine.newBuilder()
                .maximumSize(1_000) // reasonable bounds
                .build(token -> new EncodingSerializerRegistry<>(defaultEncoding, token));
        this.deserializers = Caffeine.newBuilder()
                .maximumSize(1_000) // reasonable bounds
                .build(token -> new EncodingDeserializerRegistry<>(
                        encodingsSortedByWeight, errorDecoder, emptyContainerDeserializer, token));
```

Reply:
I need to do some validation on the delta between what's retained by the objectmapper and what's retained by this class.

I had considered using time-based retention as well, which would help with the frequently-rebuild-client case, an upper bound on cache size is always good idea as well.

Reply:
created https://github.com/palantir/dialogue/pull/1587 for thoughts